app/assets/javascripts/discourse/app/components/user-menu/notifications-list.js

```diff
@@ -56,14 +53,25 @@ export default class UserMenuNotificationsList extends UserMenuItemsList {
     };
 
     const types = this.filterByTypes;
+
     if (types?.length > 0) {
       params.filter_by_types = types.join(",");
       params.silent = true;
     }
+
     return this.store
       .findStale("notification", params)
       .refresh()
-      .then((c) => c.content);
+      .then((c) => {
+        return c.content.map((notification) => {
+          return new UserMenuNotificationItem({
```

Comment:
This is the biggest advantage of this change where we require the object being returned here to conform to an interface and thus reduces the need for plugins to have to create their own component.